### ISSUE TITLE
What is the impact of DataLoader's num_worker when batching is done by Webdataset?

### ISSUE THREAD

**Issue by la-cruche** [webdataset/webdataset#151]
Hi,

I'm confused by how Webdataset goes from N workers parsing each k shards to a b-size batch:

- I see here that it is recommended to let batching be done by Webdataset instead of by  PyTorch DataLoader https://webdataset.github.io/webdataset/gettingstarted/

- I also see [here](https://github.com/webdataset/webdataset/issues/71) that Webdataset creates batches within a worker - _"you construct batches within the workers"_

So, if a batch from a Webdataset is created by only one worker, what is the point of using num_workers in the dataloader? to have N workers creating N batches at the same time? Then, what happens if in the dataloader num_workers > 1 but prefetch_factor = 1 ?

**Comment by tmbdev** [comment]
The purpose of having `num_workers > 1` is efficiency: image decompression and data augmentation are compute intensive and often require multiple cores. You gain that efficiency even with `prefetch_factor = 1`, though higher prefetch factors are probably better.

Note that you often want to shuffle data between different shards, so the recommended way of loading data is with:

```Python
dataset = WebDataset(...). ... .batched(ds_batch_size)
dataloader = WebDataloader(...).unbatched().shuffle(1000).batched(batch_size)
```
